### ISSUE TITLE
File rotation problem with large logfiles

### ISSUE THREAD

**Issue by hajgato** [easybuilders/easybuild-framework#1094]
In cases where the logfile is large  the following errors encountered:

```
Traceback (most recent call last):
  File "/usr/lib64/python2.6/logging/handlers.py", line 77, in emit
    self.doRollover()
  File "/usr/lib64/python2.6/logging/handlers.py", line 134, in doRollover
    os.rename(self.baseFilename, dfn)
OSError: [Errno 13] Permission denied
Traceback (most recent call last):
  File "/usr/lib64/python2.6/logging/handlers.py", line 76, in emit
    if self.shouldRollover(record):
  File "/usr/lib64/python2.6/logging/handlers.py", line 150, in shouldRollover
    self.stream.seek(0, 2)  #due to non-posix-compliant Windows feature
```

As a consequence, most of the log are lost, only partial log is reatined in the software/[SOFTWARE]/easybuild directory. 

There are some indications that this happens if the easybuild_run_cmd log becomes too big, and EB attaches it to the main log. 

maybe the error occours when the easybuild_run_cmd being appended to the main log, and due its size, EB tries to rotate the main log. 

More info about the python error (maybe its misleading, though): http://bugs.python.org/issue4749

**Comment by boegel** [comment]
A related issue is fixed in #1238, but the problem reported here is that the log is being rotated in the middle of a running `run_cmd`.

Switching to the `run*` methods provided by `vsc.utils.run` should fix this.

**Comment by garvct** [comment]
I got the same error, attempting to build Parallel Paraview 4.3.1 with easybuild 2.2.0

**Comment by boegel** [comment]
@garvct: thanks for bringing this up; moving to the `run*` functions provided by `vsc.utils.run` is no easy change however...

**Comment by boegel** [comment]
@garvct: do you need a custom easyblock for Parallel Paraview, or does `CMakeMake` suffice (like it does for the Paraview easyconfigs included in EB v2.2.0)?

What you may consider is to cook up a custom easyblock that used the appropriate `run` method from `vsc.utils.run` rather than `run_cmd` from `easybuild.tools.run`...

**Comment by garvct** [comment]
Kenneth,

Here is my easyconfig file (To build parallel Paraview (with pvbatch)).


easyblock = 'CMakeMake'

name = 'Paraview-mpi'
version = '4.3.1'

homepage = "http://www.paraview.org"
description = "Paraview is a scientific parallel visualizer"

toolchain = {'name': 'goolf', 'version': '4.5.0'}
toolchainopts = {'usempi': True, 'pic': True}

buildininstalldir = True

download_suffix =
'download.php?submit=Download&version=v%(version_major_minor)s&type=source&os=all&downloadFile='
source_urls = ['http://www.paraview.org/paraview-downloads/%s' %
download_suffix]
sources = ["ParaView-v%(version)s-source.tar.gz"]

#dependencies = [('Qt','4.8.6')]
comp = ('GCC','4.8.3')

dependencies = [
                ('Mesa', '7.11.2'),
                ('Python', '2.7.8')
               ]
builddependencies = [('CMake','3.2.3','',comp)
                    ]

parallel = 8

separate_build_dir = True

configopts =  "-DPARAVIEW_BUILD_QT_GUI=OFF
-DOPENGL_INCLUDE_DIR=$EBROOTMESA/include
-DOPENGL_gl_LIBRARY=$EBROOTMESA/lib/libGL.so "
configopts += "-DOPENGL_glu_LIBRARY=$EBROOTMESA/lib/libGLU.so
-DVTK_OPENGL_HAS_OSMESA=ON "
configopts += "-DOSMESA_INCLUDE_DIR=$EBROOTMESA/include
-DOSMESA_LIBRARY=$EBROOTMESA/lib/libOSMesa.so  -DVTK_USE_X=OFF "
configopts += "-DPARAVIEW_USE_MPI=ON -DPARAVIEW_USE_ICE_T=ON
-DCMAKE_BUILD_TYPE=Release -DCMAKE_VERBOSE=ON -DCMAKE_CXX_FLAGS='-fPIC -O2
-march=native -pthread' "
configopts  += "-DPARAVIEW_ENABLE_PYTHON=ON"

modextrapaths = {'LD_LIBRARY_PATH': ['EBROOTPARAVIEWMINMPI/lib/paraview-%s'
% version[:-2]],
                 'PYTHONPATH':
['EBROOTPARAVIEWMINMPI/lib/paraview-%s/site-packages' % version[:-2]]}

moduleclass = 'vis'


Thanks,
Cormac.

On Mon, Aug 17, 2015 at 12:16 PM, Kenneth Hoste <notifications@github.com>
wrote:

> @garvct <https://github.com/garvct>: thanks for bringing this up; moving
> to the run* functions provided by vsc.utils.run is no easy change
> however...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/hpcugent/easybuild-framework/issues/1094#issuecomment-131913965>
> .
>



-- 
Cormac Garvey
HPC Software Consultant
Scientific Computing
Idaho National Laboratory
Ph: 208-526-6294

**Comment by gppezzi** [comment]
We also get this error message on Piz Daint when building paraview.

The rotated log files seem to be there after the build, except for these error messages (that are displayed only at the console and aren't captured on the log).

**Comment by holstgr-kaust** [comment]
This is still occurring in EasyBuild v3.1.2, also with a custom ParaView (v5.3.0) build config.  The behaviour is exactly as described above:

```
Traceback (most recent call last):
  File "/usr/lib64/python2.7/logging/handlers.py", line 77, in emit
    self.doRollover()
  File "/usr/lib64/python2.7/logging/handlers.py", line 140, in doRollover
    os.rename(self.baseFilename, dfn)
OSError: [Errno 13] Permission denied
Logged from file build_log.py, line 147
```

That ParaView config is building more components than previous working ParaView configs had.

Maybe log rotation should be disabled (e.g., set MAX_BYTES = 0 in vsc.utils.fancylogger) until it works reliably?

**Comment by holstgr-kaust** [comment]
This issue: https://github.com/hpcugent/vsc-base/issues/252
Has a workaround hack to fix issue.

**Comment by boegel** [comment]
I consider this fixed via #2347, thanks for bringing attention to this again @holstgr-kaust!
